## Search Intelligence Data Analysis Program 
---

In [238]:
import pandas as pd 

q1df = pd.read_csv("Question - 1.csv")

q2dfrap = pd.read_csv("Q2 Dataframes/Question - 2 rap.csv")
q2dfrock = pd.read_csv("Q2 Dataframes/Question - 2 rock.csv")
q2dfstates = pd.read_csv("Q2 Dataframes/Question - 2 states.csv")
# imports the necessary libraries

In [239]:
q1df.head() 
# displays dataframe

Capital          State  Meal, Inexpensive Restaurant  \
0   Frankfort       Kentucky                          13.0   
1  Montgomery        Alabama                          17.0   
2      Topeka         Kansas                          13.5   
3  Charleston  West Virginia                          13.0   
4     Lansing       Michigan                          15.0   

   Meal for 2 People, Mid-range Restaurant, Three-course  \
0                                               48.0       
1                                               50.0       
2                                               54.0       
3                                               40.0       
4                                               70.0       

   Mcmeal at Mcdonalds Or Equivalent Combo Meal  Domestic Beer 1 Pint Draught  \
0                                           8.0                          3.25   
1                                          10.0                          4.00   
2                                          10.0                          5.00   
3                                           8.0                          4.00   
4                                          10.0                          4.00   

   Imported Beer 12 Oz Small Bottlein Restaurant  Cappuccino Regular  \
0                                           8.00                4.33   
1                                           3.75                4.50   
2                                           6.00                3.50   
3                                           6.00                2.50   
4                                           6.00                4.72   

   Coke/pepsi 12 Oz Small Bottle  Water 12 Oz Small Bottle  ...  \
0                           1.77                      1.29  ...   
1                           2.00                      2.12  ...   
2                           2.08                      1.56  ...   
3                           1.30                      1.51  ...   
4                           2.19                      1.68  ...   

   Rank - Childcare  Rank - Clothing  Rank - Cost to Buy Apartment  \
0                 1                3                            23   
1                 2                8                             2   
2                 5                2                             3   
3                23               25                             1   
4                16               16                             5   

   Rank - Leisure  Rank - Markets  Rank - Monthly Rent  \
0               6               2                    2   
1               1               1                    3   
2               7               8                    8   
3              12               6                   16   
4               2              10                    4   

   Rank - Monthly Utilities  Rank - Restaurants  \
0                        27                   2   
1                        11                   9   
2                        12                   7   
3                         2                   1   
4                         9                  12   

   Rank - Salaries and Financing  Rank - Transportation  
0                             21                     15  
1                             26                     32  
2                             13                      3  
3                             45                     14  
4                             36                     12  

[5 rows x 74 columns]

**Question 1**

---

In [240]:
numeric_columns = q1df.columns.difference(['Capital', 'State'])
q1df[numeric_columns] = q1df[numeric_columns].apply(pd.to_numeric, errors='coerce')
# Converts every row aside from the initial name to a numerical value 

q1df['Total Cost'] = q1df[numeric_columns].sum(axis=1)
# Calculates the sum of each numerical row

least_expensive = q1df.loc[q1df['Total Cost'].idxmin()]
# Finds the lowest result and saves it under a new variable 

print("The least expensive capital state to live in is:")
print(least_expensive[['Capital', 'State', 'Total Cost']]) 
# Displays the findings 

# This code can be altered to find other instances of data such as the most expensive by changing 'idxmin' to 'idxman' etc.

The least expensive capital state to live in is:
Capital          Dover
State         Delaware
Total Cost     1574.29
Name: 31, dtype: object


> It is shown that Dover is the least expensive state with a total cost of $1574.29.

**Question 2**

---

In [241]:
q2dfrap.head()
# Shows highest ranking rap artist 

Rap Avg. Monthly Volume  Rank
0   Post Malone              16,258     1
1  XXXTentacion              14,821     2
2      Lil Peep              13,563     3
3      Doja Cat              13,386     4
4       Cardi B              12,168     5

In [242]:
q2dfrock.head()
# Shows highest ranking rock artist 

Rock Avg. Monthly Volume  Rank
0  Billie Eilish              23,484     1
1        Rihanna               9,204     2
2        Chicago               6,847     3
3      Metallica               3,035     4
4        Nirvana               2,928     5

In [243]:
sortedq2df = q2dfstates.sort_values('Keyword Searches (per 100k Population)', ascending=[False])
sortedq2df.head(6)
# Shows the state with the most searched keyword 

Location  Population Avg. Monthly Keyword Searches  \
0      Illinois  12,549,689                        22,386   
1       Wyoming     584,057                           956   
2        Nevada   3,194,176                         5,039   
3      Delaware   1,031,890                         1,628   
4  North Dakota     783,926                         1,205   
5  Rhode Island   1,095,962                         1,672   

   Keyword Searches (per 100k Population)  Ranking (per 100k Population)  
0                                  178.38                              1  
1                                  163.65                              2  
2                                  157.76                              3  
3                                  157.72                              4  
4                                  153.71                              5  
5                                  152.53                              6

> From the rankings, Billie Eilish is the most searched rock artist and Post Malone is the most searched rap artist with Billie Eilish having the highest searches overall. Additionally, the most amount of tattoo searches per state is within Illinois at 178.38 per 100k.

**Question 4**

---

In [248]:
# Due to the nature of the CSV file, it is easier to recreate the tables
Scarborough = {
    "Parking Spot": [
        "Dean Road", "Dean Road (2)", "Scarborough Station", "Albion Road",
        "Queens Parade", "Adelaide House", "Somerset Terrace", "Pavillion Square",
        "Falcolners Road", "Palm Court Hotel", "Friarsway", "Brunswick Shopping Centre",
        "Balmoral", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£2.80", "£3.80", "£4.20", "£6.00", "£6.45", "£6.90", "£7.00",
        "£8.00", "£8.50", "£8.70", "£9.00", "£9.70", "£10.95", "£7.08"
    ]
}
Scarborough = pd.DataFrame(Scarborough)
Scarborough = Scarborough[Scarborough["Parking Spot"] != "Average:"].reset_index(drop=True)
Scarborough["Cost (8 hour basis) (£)"] = Scarborough["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
# 
Whitby = {
    "Parking Spot": [
        "Silver Street", "St Hildas Terrace", "Cliff Street",
        "Church Street", "Whitby Station", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£6.00", "£8.00", "£8.50", "£9.00", "£10.00", "£8.30"
    ]
}
Whitby = pd.DataFrame(Whitby)
Whitby = Whitby[Whitby["Parking Spot"] != "Average:"].reset_index(drop=True)
Whitby["Cost (8 hour basis) (£)"] = Whitby["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Blackpool = {
    "Parking Spot": [
        "Buchanan Street", "Hornby Road", "Caunce Street", "Whitegate Drive",
        "Reads Avenue", "General Street", "Cocker Street", "Jaggy Thistle",
        "Metropole Hotel", "Queen Street", "Dickson Road", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£3.50", "£5.70", "£5.90", "£5.95", "£6.00", "£6.50", "£7.00",
        "£8.00", "£9.00", "£11.00", "£14.95", "£7.59"
    ]
}
Blackpool = pd.DataFrame(Blackpool)
Blackpool = Blackpool[Blackpool["Parking Spot"] != "Average:"].reset_index(drop=True)
Blackpool["Cost (8 hour basis) (£)"] = Blackpool["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
# 
Hornsea = {
    "Parking Spot": [
        "Morrow Avenue", "Hornsea Burton Road", "Eastgate", "Broadway", 
        "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£0.00", "£0.00", "£3.30", "£4.60", "£1.98"
    ]
}

Hornsea = pd.DataFrame(Hornsea)
Hornsea = Hornsea[Hornsea["Parking Spot"] != "Average:"].reset_index(drop=True)
Hornsea["Cost (8 hour basis) (£)"] = Hornsea["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Staithes = {
    "Parking Spot": [
        "Staithes Lane", "Bank Top", "Cowbar Lane", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£5.80", "£6.00", "£6.50", "£6.10"
    ]
}
Staithes = pd.DataFrame(Staithes)
Staithes = Staithes[Staithes["Parking Spot"] != "Average:"].reset_index(drop=True)
Staithes["Cost (8 hour basis) (£)"] = Staithes["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Morecambe = {
    "Parking Spot": [
        "Back Brighton Terrace", "Victoria Street", "Billy Hill", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£2.00", "£3.00", "£8.00", "£4.33"
    ]
}
Morecambe = pd.DataFrame(Morecambe)
Morecambe = Morecambe[Morecambe["Parking Spot"] != "Average:"].reset_index(drop=True)
Morecambe["Cost (8 hour basis) (£)"] = Morecambe["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Filey = {
    "Parking Spot": [
        "Filey Station", "Station Avenue", "Coble Landing", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£1.00", "£8.00", "£9.00", "£6.00"
    ]
}
Filey = pd.DataFrame(Filey)
Filey = Filey[Filey["Parking Spot"] != "Average:"].reset_index(drop=True)
Filey["Cost (8 hour basis) (£)"] = Filey["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Bridlington = {
    "Parking Spot": [
        "Stepney Grove", "York Road", "Moorfield Road", "Bridlington Station",
        "Nelson Street", "Langdales Wharf", "The Promenade", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£0.00", "£2.50", "£3.30", "£4.00", "£4.50", "£6.30", "£11.60", "£4.60"
    ]
}
Bridlington = pd.DataFrame(Bridlington)
Bridlington = Bridlington[Bridlington["Parking Spot"] != "Average:"].reset_index(drop=True)
Bridlington["Cost (8 hour basis) (£)"] = Bridlington["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Brighton = {
    "Parking Spot": [
        "Chapel Street", "Brighton Station", "Trafalgar Street", "Churchill Square",
        "Regency Square", "The Lanes", "North Road", "Theatre",
        "Brighton Centre Russel Road", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£10.00", "£14.90", "£18.90", "£20.00", "£22.80", "£31.40",
        "£32.00", "£32.95", "£34.95", "£24.21"
    ]
}
Brighton = pd.DataFrame(Brighton)
Brighton = Brighton[Brighton["Parking Spot"] != "Average:"].reset_index(drop=True)
Brighton["Cost (8 hour basis) (£)"] = Brighton["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Eastbourne = {
    "Parking Spot": [
        "Devonshire Park", "Eastbourne Station", "Trinity Place", "The Beacon",
        "Wish Tower", "Enterprise Shopping Centre", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£4.80", "£6.00", "£7.95", "£9.60", "£10.00", "£16.00", "£9.06"
    ]
}
Eastbourne = pd.DataFrame(Eastbourne)
Eastbourne = Eastbourne[Eastbourne["Parking Spot"] != "Average:"].reset_index(drop=True)
Eastbourne["Cost (8 hour basis) (£)"] = Eastbourne["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Bournemouth = {
    "Parking Spot": [
        "Durley", "Central", "Exeter Park Road", "Bh2", "The Claremont Hotel",
        "New Westcliff Hotel", "Winter Gardens", "Terrace Road", 
        "The Square", "Bournemouth International Centre", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£5.00", "£5.80", "£7.00", "£8.50", "£8.95", "£10.00", 
        "£11.50", "£14.95", "£15.95", "£16.50", "£10.42"
    ]
}
Bournemouth = pd.DataFrame(Bournemouth)
Bournemouth = Bournemouth[Bournemouth["Parking Spot"] != "Average:"].reset_index(drop=True)
Bournemouth["Cost (8 hour basis) (£)"] = Bournemouth["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Dover = {
    "Parking Spot": [
        "St Martins Battery", "Dover Priory Station", "Albany Place",
        "Camden Crescent", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£0.00", "£4.80", "£7.80", "£9.60", "£5.55"
    ]
}
Dover = pd.DataFrame(Dover)
Dover = Dover[Dover["Parking Spot"] != "Average:"].reset_index(drop=True)
Dover["Cost (8 hour basis) (£)"] = Dover["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Margate = {
    "Parking Spot": [
        "Mill Lane", "Morrisons", "Turner Contemporary", 
        "Trinity Square", "Dreamland", "Poundstretcher", 
        "Harbour Arm", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£0.00", "£6.00", "£8.00", "£8.20", "£9.00", 
        "£10.00", "£12.00", "£7.60"
    ]
}
Margate = pd.DataFrame(Margate)
Margate = Margate[Margate["Parking Spot"] != "Average:"].reset_index(drop=True)
Margate["Cost (8 hour basis) (£)"] = Margate["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Deal = {
    "Parking Spot": [
        "Deal Station", "Union Road", "Beach Street", 
        "Tides", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£3.80", "£7.80", "£9.60", "£7.00", "£7.05"
    ]
}
Deal = pd.DataFrame(Deal)
Deal = Deal[Deal["Parking Spot"] != "Average:"].reset_index(drop=True)
Deal["Cost (8 hour basis) (£)"] = Deal["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Southend_on_Sea = {
    "Parking Spot": [
        "Southend Victoria Station", "South Essex College", "Guildford Road", 
        "Victoria", "Central Station", "Warrior Square", 
        "Southchurch Avenue", "Seaway", "Royals Shopping", 
        "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£5.00", "£6.70", "£8.00", "£8.30", "£9.30", 
        "£13.20", "£13.73", "£19.30", "£20.00", "£11.50"
    ]
}
Southend_on_Sea = pd.DataFrame(Southend_on_Sea)
Southend_on_Sea = Southend_on_Sea[Southend_on_Sea["Parking Spot"] != "Average:"].reset_index(drop=True)
Southend_on_Sea["Cost (8 hour basis) (£)"] = Southend_on_Sea["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Ilfracombe = {
    "Parking Spot": [
        "Brookdale", "Ropery Road", "High Street", 
        "Wilder Road", "Jubilee Gardens", "Marine Drive", 
        "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£6.00", "£9.50", "£10.00", "£10.40", 
        "£12.00", "£12.10", "£10.00"
    ]
}
Ilfracombe = pd.DataFrame(Ilfracombe)
Ilfracombe = Ilfracombe[Ilfracombe["Parking Spot"] != "Average:"].reset_index(drop=True)
Ilfracombe["Cost (8 hour basis) (£)"] = Ilfracombe["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Burnham_on_Sea = {
    "Parking Spot": [
        "Discount", "Oxford Street", "High Street", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£4.40", "£5.50", "£6.10", "£5.33"
    ]
}
Burnham_on_Sea = pd.DataFrame(Burnham_on_Sea)
Burnham_on_Sea = Burnham_on_Sea[Burnham_on_Sea["Parking Spot"] != "Average:"].reset_index(drop=True)
Burnham_on_Sea["Cost (8 hour basis) (£)"] = Burnham_on_Sea["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Weston_Super_Mare = {
    "Parking Spot": [
        "Walliscote Place", "Carlton Street", "APCOA Parking", 
        "Locking Road Long Stay", "Sovreign Shopping Centre", 
        "High Street", "North Somerset Council", "Grove Park", 
        "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£3.00", "£5.00", "£5.40", "£5.50", 
        "£6.00", "£9.95", "£10.00", "£12.00", 
        "£7.11"
    ]
}
Weston_Super_Mare = pd.DataFrame(Weston_Super_Mare)
Weston_Super_Mare = Weston_Super_Mare[Weston_Super_Mare["Parking Spot"] != "Average:"].reset_index(drop=True)
Weston_Super_Mare["Cost (8 hour basis) (£)"] = Weston_Super_Mare["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Torquay = {
    "Parking Spot": [
        "Daddyhole Plain", "2 Temperance Street", "Inglewood Palm Hotel", 
        "Melville Street", "Torquay Marina", "Harbour", 
        "Fleet Walk Shopping Centre", "Beacon Quay", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£0.00", "£6.50", "£8.00", "£9.30", 
        "£9.50", "£9.80", "£10.00", "£10.50", 
        "£7.95"
    ]
}
Torquay = pd.DataFrame(Torquay)
Torquay = Torquay[Torquay["Parking Spot"] != "Average:"].reset_index(drop=True)
Torquay["Cost (8 hour basis) (£)"] = Torquay["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
St_Ives = {
    "Parking Spot": [
        "Smeatons Pier", "Island", "The Stennack", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£9.00", "£10.00", "£12.00", "£10.33"
    ]
}
St_Ives = pd.DataFrame(St_Ives)
St_Ives = St_Ives[St_Ives["Parking Spot"] != "Average:"].reset_index(drop=True)
St_Ives["Cost (8 hour basis) (£)"] = St_Ives["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Newquay = {
    "Parking Spot": [
        "Edgecumbe Avenue", "Tregunnel Hill", "South Fistral", 
        "Most Holy Trinity Church", "Towan Headland", "Tower Road", 
        "Feel Good Building", "Newquay Station", "Fistral Beach", 
        "Hotel Victoria", "Town View", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£5.95", "£6.00", "£6.40", "£8.50", 
        "£8.70", "£9.50", "£10.00", "£10.50", 
        "£12.00", "£25.00", "£33.00", "£12.32"
    ]
}
Newquay = pd.DataFrame(Newquay)
Newquay = Newquay[Newquay["Parking Spot"] != "Average:"].reset_index(drop=True)
Newquay["Cost (8 hour basis) (£)"] = Newquay["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Perranporth = {
    "Parking Spot": [
        "Ponsmere Valley", "Droskyn", "The Green Parrot", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£5.00", "£5.50", "£10.00", "£6.83"
    ]
}
Perranporth = pd.DataFrame(Perranporth)
Perranporth = Perranporth[Perranporth["Parking Spot"] != "Average:"].reset_index(drop=True)
Perranporth["Cost (8 hour basis) (£)"] = Perranporth["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Padstow = {
    "Parking Spot": [
        "South Quay", "Lawns", "Link Road", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£5.50", "£8.00", "£10.00", "£7.83"
    ]
}
Padstow = pd.DataFrame(Padstow)
Padstow = Padstow[Padstow["Parking Spot"] != "Average:"].reset_index(drop=True)
Padstow["Cost (8 hour basis) (£)"] = Padstow["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Falmouth = {
    "Parking Spot": [
        "Penmere Station", "Discovery Quay", "Church Street", 
        "Custom House Quay", "Maritime", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£0.00", "£5.00", "£10.00", "£10.40", 
        "£35.20", "£12.12"
    ]
}
Falmouth = pd.DataFrame(Falmouth)
Falmouth = Falmouth[Falmouth["Parking Spot"] != "Average:"].reset_index(drop=True)
Falmouth["Cost (8 hour basis) (£)"] = Falmouth["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Portsmouth = {
    "Parking Spot": [
        "Fratton Station", "Victory Retail Park", "Crasswell Street", 
        "Clarence Street", "Old Commercial Road", "Market Way", 
        "Ferry House Lodge", "All Saints", "Portsmouth International Port", 
        "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£3.90", "£5.00", "£6.20", "£8.40", 
        "£8.80", "£9.95", "£10.00", "£12.40", 
        "£13.00", "£8.63"
    ]
}
Portsmouth = pd.DataFrame(Portsmouth)
Portsmouth = Portsmouth[Portsmouth["Parking Spot"] != "Average:"].reset_index(drop=True)
Portsmouth["Cost (8 hour basis) (£)"] = Portsmouth["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Tenby = {
    "Parking Spot": [
        "Tenby Leisure Centre", "Tenby Station", "North Beach", 
        "The Royal Gatehouse", "Salterns", "Seafront", 
        "Five Arches", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£0.00", "£4.25", "£5.00", "£6.00", 
        "£7.00", "£8.00", "£10.00", "£5.75"
    ]
}
Tenby = pd.DataFrame(Tenby)
Tenby = Tenby[Tenby["Parking Spot"] != "Average:"].reset_index(drop=True)
Tenby["Cost (8 hour basis) (£)"] = Tenby["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Llandudno = {
    "Parking Spot": [
        "Llandudno Station", "Llandudno FC", "Maelgwyn Road", 
        "The Victoria Centre", "Venue Cymru", "Parc Llandudno Retail Park", 
        "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£2.00", "£5.00", "£5.60", "£6.00", 
        "£7.50", "£12.00", "£6.35"
    ]
}
Llandudno = pd.DataFrame(Llandudno)
Llandudno = Llandudno[Llandudno["Parking Spot"] != "Average:"].reset_index(drop=True)
Llandudno["Cost (8 hour basis) (£)"] = Llandudno["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Colwyn_Bay = {
    "Parking Spot": [
        "Colwyn Bay Station", "Douglas Road", "Ivy Street", 
        "Bayview Shopping Centre", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£4.75", "£5.00", "£5.60", "£8.00", 
        "£5.84"
    ]
}
Colwyn_Bay = pd.DataFrame(Colwyn_Bay)
Colwyn_Bay = Colwyn_Bay[Colwyn_Bay["Parking Spot"] != "Average:"].reset_index(drop=True)
Colwyn_Bay["Cost (8 hour basis) (£)"] = Colwyn_Bay["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Fife = {
    "Parking Spot": [
        "107 North St", "Argyle Street North", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£0.00", "£3.80", "£1.90"
    ]
}
Fife = pd.DataFrame(Fife)
Fife = Fife[Fife["Parking Spot"] != "Average:"].reset_index(drop=True)
Fife["Cost (8 hour basis) (£)"] = Fife["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)
#
Portrush = {
    "Parking Spot": [
        "Fire Station", "Dunluce Avenue", "Harbour Road", "Average:"
    ],
    "Cost (8 hour basis) (£)": [
        "£0.00", "£2.40", "£8.00", "£3.47"
    ]
}
Portrush = pd.DataFrame(Portrush)
Portrush = Portrush[Portrush["Parking Spot"] != "Average:"].reset_index(drop=True)
Portrush["Cost (8 hour basis) (£)"] = Portrush["Cost (8 hour basis) (£)"].replace({'£': ''}, regex=True).astype(float)

In [254]:
dataframes = {
    "Scarborough": Scarborough,
    "Whitby": Whitby,
    "Blackpool": Blackpool,
    "Hornsea": Hornsea,
    "Staithes": Staithes,
    "Morecambe": Morecambe,
    "Filey": Filey,
    "Bridlington": Bridlington,
    "Brighton": Brighton,
    "Eastbourne": Eastbourne,
    "Bournemouth": Bournemouth,
    "Dover": Dover,
    "Margate": Margate,
    "Deal": Deal,
    "Southend_on_Sea": Southend_on_Sea,
    "Ilfracombe": Ilfracombe,
    "Burnham_on_Sea": Burnham_on_Sea,
    "Weston_Super_Mare": Weston_Super_Mare,
    "Torquay": Torquay,
    "St_Ives": St_Ives,
    "Newquay": Newquay,
    "Perranporth": Perranporth,
    "Padstow": Padstow,
    "Falmouth": Falmouth,
    "Portsmouth": Portsmouth,
    "Tenby": Tenby,
    "Llandudno": Llandudno,
    "Colwyn_Bay": Colwyn_Bay,
    "Fife": Fife,
    "Portrush": Portrush
}

summary_stats = pd.DataFrame({
    "Mean": [df["Cost (8 hour basis) (£)"].mean() for df in dataframes.values()],
    "Median": [df["Cost (8 hour basis) (£)"].median() for df in dataframes.values()],
    "Min": [df["Cost (8 hour basis) (£)"].min() for df in dataframes.values()],
    "Max": [df["Cost (8 hour basis) (£)"].max() for df in dataframes.values()]
}, index=dataframes.keys())

sorted_summary_stats = summary_stats.sort_values(by='Median', ascending=False)
print(sorted_summary_stats)



                        Mean  Median    Min    Max
Brighton           24.211111  22.800  10.00  34.95
Ilfracombe         10.000000  10.200   6.00  12.10
St_Ives            10.333333  10.000   9.00  12.00
Falmouth           12.120000  10.000   0.00  35.20
Newquay            12.322727   9.500   5.95  33.00
Bournemouth        10.415000   9.475   5.00  16.50
Torquay             7.950000   9.400   0.00  10.50
Southend_on_Sea    11.503333   9.300   5.00  20.00
Portsmouth          8.627778   8.800   3.90  13.00
Eastbourne          9.058333   8.775   4.80  16.00
Whitby              8.300000   8.500   6.00  10.00
Margate             7.600000   8.200   0.00  12.00
Padstow             7.833333   8.000   5.50  10.00
Filey               6.000000   8.000   1.00   9.00
Deal                7.050000   7.400   3.80   9.60
Scarborough         7.076923   7.000   2.80  10.95
Blackpool           7.590909   6.500   3.50  14.95
Dover               5.550000   6.300   0.00   9.60
Tenby               5.750000   